src/boot/boot.c

```diff
@@ -87,7 +87,7 @@ typedef struct {
         bool options_implied; /* If true, these options are implied if we invoke the PE binary without any parameters (as in: UKI). If false we must specify these options explicitly. */
         char16_t **initrd;
         char16_t key;
-        EFI_STATUS (*call)(void);
+        EFI_STATUS (*call)(const struct BootEntry *entry, EFI_FILE *root_dir, EFI_HANDLE parent_image);
```

Comment:
Please make the argument consistent to other places: `root_dir` -> `root`

Reply:
uh, it's actually all over the place. Some EFI_FILE* objects are called "root", others are "root_dir", for the same concepts... not sure i want to clean that up now. 

i'll make sure that for the stuff I am adding here i'll make it systematic. I think root_dir is more common, will stick to that.

Reply:
Then better to rename all `root` -> `root_dir` for `call` functions introduced in this PR?